sdks/java/io/redis/src/test/java/org/apache/beam/sdk/io/redis/RedisIOTest.java

```diff
@@ -250,6 +260,102 @@ public void testWriteUsingDECRBY() {
     assertEquals(-1, count);
   }
 
+  @Test
+  public void testWriteStreams() {
+    ArrayList<String> streams = new ArrayList<String>();
+    for (int i = 0; i <= 10; i++) {
+      UUID uuid = UUID.randomUUID();
+      /* stream keys are uuids to ensure that test runs are idempotent */
+      streams.add(uuid.toString());
+    }
+    Map<String, String> fooValues = ImmutableMap.of("sensor-id", "1234", "temperature", "19.8");
+    Map<String, String> barValues = ImmutableMap.of("sensor-id", "9999", "temperature", "18.2");
+
+    List<KV<String, Map<String, String>>> fooData =
+        streams.stream().map(key -> KV.of(key, fooValues)).collect(Collectors.toList());
+
+    List<KV<String, Map<String, String>>> barData =
+        streams.stream().map(key -> KV.of(key, barValues)).collect(Collectors.toList());
+
+    List<KV<String, Map<String, String>>> allData =
+        Stream.of(fooData, barData).flatMap(Collection::stream).collect(Collectors.toList());
```

Comment:
Just pointing out a more idiomatic  alternative here, again using streams to generate the input data (though not blocking)
```java
    Map<String, String> fooValues = ImmutableMap.of("sensor-id", "1234", "temperature", "19.8");
    Map<String, String> barValues = ImmutableMap.of("sensor-id", "9999", "temperature", "18.2");

    List<KV<String, Map<String, String>>> allData = IntStream.range(0, 10)
        .boxed()
        .flatMap(id -> Stream.of(
            KV.of("testWriteStreams"+id, fooValues),
            KV.of("testWriteStreams"+id, barValues)))
        .collect(toList());
```

Reply:
Aha, thank you, I knew there had to be a simpler way to do this.  I've made the slight adjustment of preserving the randomized key names (otherwise test runs end up not necessarily being idempotent) but this is much more readable.